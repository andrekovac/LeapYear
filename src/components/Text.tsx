import { Animated } from "react-native";
import styled from "styled-components/native";

const Text = styled(Animated.Text)<{ size?: number }>`
  font-size: ${(props) => (props.size ? `${props.size}px` : "30px")};
  font-weight: 100;
  text-align: center;
`;

export default Text;
