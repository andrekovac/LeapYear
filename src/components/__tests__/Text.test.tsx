import React from "react";
import renderer from "react-test-renderer";

import Text from "../Text";

describe("Text", () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(<Text />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders correctly with size prop`, () => {
    const tree = renderer.create(<Text size={100} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
