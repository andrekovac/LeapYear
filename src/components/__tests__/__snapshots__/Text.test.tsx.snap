// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "black",
      "fontSize": 30,
      "fontWeight": "100",
      "textAlign": "center",
    }
  }
/>
`;

exports[`Text renders correctly with size prop 1`] = `
<Text
  size={100}
  style={
    Object {
      "color": "black",
      "fontSize": 100,
      "fontWeight": "100",
      "textAlign": "center",
    }
  }
/>
`;
